sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HistogramNumeric.scala

```diff
@@ -125,7 +125,21 @@ case class HistogramNumeric(
     } else {
       val result = (0 until buffer.getUsedBins).map { index =>
         val coord = buffer.getBin(index)
-        InternalRow.apply(coord.x, coord.y)
+        // The 'coord.x' and 'coord.y' have double type here. If the expression is expected to
+        // return a result of double type, this is sufficient. Otherwise, we need to internally
+        // convert these values to the expected result type, for cases like timestamps and intervals
+        // which are valid inputs to the numeric histogram aggregate function.
+        val result: Any = left.dataType match {
+          case ByteType => coord.x.toByte
+          case IntegerType | DateType | _: YearMonthIntervalType =>
+            coord.x.toInt
```

Comment:
I think we should explain this behaviour in the documentation above L54 with an example.